idc_index/index.py

```diff
@@ -502,6 +465,364 @@ def get_viewer_URL(
+            how="left",
+        )
+        sync_size = merged_df["series_size_MB"].sum()
+        sync_size_rounded = round(sync_size, 2)
+        # Write a temporary manifest file
+        with tempfile.NamedTemporaryFile(mode="w", delete=False) as synced_manifest:
+            for s3_url in merged_df["s3_url"]:
+                synced_manifest.write(f"sync {s3_url} {downloadDir}\n")
+        return Path(synced_manifest.name), sync_size_rounded
+
+    def _s5cmd_run(
+        self, endpoint_to_use, manifest_file, total_size, downloadDir, quiet=False
+    ):
+        """
+        Executes the s5cmd command to sync files from a given endpoint to a local directory.
+
+        This function first performs a dry run of the s5cmd command to check which files need to be downloaded.
+        If there are files to be downloaded, it generates a new manifest file with the files to be synced and
+        runs the s5cmd command again to download the files. The progress of the download is tracked and printed
+        to the console.
+
+        Args:
+            endpoint_to_use (str): The endpoint URL to download the files from.
+            manifest_file (str): The path to the manifest file listing the files to be downloaded.
+            total_size (float): The total size of the files to be downloaded in MB.
+            downloadDir (str): The local directory where the files will be downloaded.
+            quiet (bool, optional): If True, suppresses the stdout and stderr of the s5cmd command. Defaults to False.
+
+        Raises:
+            subprocess.CalledProcessError: If the s5cmd command fails.
+
+        Returns:
+            None
+        """
+
+        dry_run_cmd = [
+            self.s5cmdPath,
+            "--no-sign-request",
+            "--dry-run",
+            "--endpoint-url",
+            endpoint_to_use,
+            "run",
+            manifest_file,
+        ]
+
+        process = subprocess.run(
+            dry_run_cmd, stdout=subprocess.PIPE, text=True, check=False
+        )
+
+        if process.stdout:
+            # Some files need to be downloaded
+            (
+                synced_manifest,
+                sync_size,
+            ) = self._parse_s5cmd_sync_output_and_generate_synced_manifest(
+                stdout=process.stdout, downloadDir=downloadDir
+            )
+            cmd = [
+                self.s5cmdPath,
+                "--no-sign-request",
+                "--endpoint-url",
+                endpoint_to_use,
+                "run",
+                synced_manifest,
+            ]
+
+            if quiet:
+                stdout = subprocess.PIPE
+                stderr = subprocess.STDOUT
+            else:
+                stdout = None
+                stderr = None
+
+            with subprocess.Popen(
+                cmd, stdout=stdout, stderr=stderr, universal_newlines=True
+            ) as process:
+                print("Total requested download size:", round(total_size, 2), " MB")
```

Comment:
I suggest adding a check here that would switch to GB if the size is above 1000 MB, and TB if the size is above 1000,000 MB to improve readability of this message.